### DISCUSSION TITLE
Boosting incorrectly applied in 6.0

### DISCUSSION THREAD

**Discussion by krivulcik** [ravendb/ravendb#16404] [Q&A] [answered]
During an investigation of another issue, I noticed that boosting seems to be applied incorrectly in 6.0.

Let's consider demo data and the following query, which searches for "some products" (`De*` here to select only two products), and then wants to display the products in two groups:
 1. Cheap products (Price <= 20)
 2. Expensive products (Price > 20).

Sorting in both groups should be by price descending.

The query looks like this:

```
from index 'Product/Search'
where
boost(search(Name, "De*"), 0)
and
(boost(PricePerUnit <= 20, 100)
or
(boost(PricePerUnit > 20, 1)))
order by score(), PricePerUnit as double desc
include explanations()
```

This works correctly in RavenDB 5. The scores for the documents are 0.499975 and 0.00499975.

However, in RavenDB 6, the scores are exactly the same for both documents: 0.35355338. The documents are thus in the same price group and not sorted according to the requirements.

![image](https://user-images.githubusercontent.com/7997055/236453223-2c83df57-c95e-4891-aabd-6dbe5efc01e4.png)

![image](https://user-images.githubusercontent.com/7997055/236452986-fb354b58-cad7-4a2a-a3ae-974481d6a24a.png)

Unable to provide a unit test as I'm investigating an issue with DocumentQuery construction.

In our real-life scenario, we use this approach to ensure null values are sorted last.

The live-test instance currently runs 6.0.0-nightly-20230425-0459.

**Comment by ayende** [comment]
Are you using a Corax or Lucene index here?

**Comment by krivulcik** [comment]
That was on live-test, so whatever is the default there.

**Reply by ayende** [reply]
Can you check if this was a sharded or non sharded db, if this was on live-test, it may be either.
Can you post the _text_ of the query here?

**Comment by maciejaszyk** [comment, marked as answer]
Hi @krivulcik,
I managed to find the root cause of the issue you have reported. 
We've enabled an experimental feature on our live server so it uses query-field-cache to optimize range queries and it seems there is a problem with appending boosting.

I created a ticket for this:
https://issues.hibernatingrhinos.com/issue/RavenDB-20443/Boosting-is-not-applied-when-QueryClauseCache-is-enabled.

This cache can be disabled via the configuration option:
`Indexing.QueryClauseCache.Disabled`

Thanks for reporting this :)
